### PR TITLE
feat: dashboard search and filter

### PR DESCRIPTION
**Pull request by pranshuchittora**
https://user-images.githubusercontent.com/32242596/164453125-09715cae-cb7b-4846-848e-d878f3dce149.mov

### CODE REVIEW

**Review by ankitnayan — APPROVED** [review]

File: frontend/src/container/ListOfDashboard/SearchFilter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
+import { CloseCircleFilled } from '@ant-design/icons';
+import { useMachine } from '@xstate/react';
+import { Button, Select } from 'antd';
+import { RefSelectProps } from 'antd/lib/select';
+import history from 'lib/history';
+import { filter, map } from 'lodash-es';
```
Comment (palash-signoz) [inline frontend/src/container/ListOfDashboard/SearchFilter/index.tsx:6]: please import via 
`import filter from "lodash-es/filter"`

also we have native filter via js

File: frontend/src/container/ListOfDashboard/SearchFilter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	convertURLQueryStringToQuery,
+	executeSearchQueries,
+	OptionsSchemas,
+	OptionsValueResolution,
+} from './utils';
+
+function SearchFilter({
+	searchData,
+	filterDashboards,
+}: {
```
Comment (palash-signoz) [inline frontend/src/container/ListOfDashboard/SearchFilter/index.tsx:29]: Please make SearchFilterProps here

File: frontend/src/container/ListOfDashboard/SearchFilter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	useEffect(() => {
+		const searchQueryString = new URLSearchParams(history.location.search).get(
+			'search',
+		);
+		if (searchQueryString)
+			setQueries(convertURLQueryStringToQuery(searchQueryString) || []);
+	}, []);
+	useEffect(() => {
+		filterDashboards(executeSearchQueries(queries, searchData));
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (palash-signoz) [inline frontend/src/container/ListOfDashboard/SearchFilter/index.tsx:52]: can you help why do we need to disable this eslint rule

File: frontend/src/container/ListOfDashboard/SearchFilter/utils.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable sonarjs/cognitive-complexity */
+import { decode, encode } from 'js-base64';
+import { flattenDeep, map, uniqWith } from 'lodash-es';
```
Comment (palash-signoz) [inline frontend/src/container/ListOfDashboard/SearchFilter/utils.ts:5]: same here

File: frontend/src/container/ListOfDashboard/SearchFilter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+				<Select
+					placeholder={
+						!queries.length &&
+						!staging.length &&
+						!selectedValues.length &&
+						'Search or Filter results'
+					}
+					size="small"
+					ref={selectRef}
+					mode={optionsData.mode as 'tags' | 'multiple'}
```
Comment (palash-signoz) [inline frontend/src/container/ListOfDashboard/SearchFilter/index.tsx:184]: Can you import this mode from SelectProps from antd

**Comment by pranay01** [comment]
@pranshuchittora Couple of comments:
1. In current implementation, users can't do free text - can we enable that
2. I think we need a search icon at the right corner of search bar - as it is not clear after multi select, when adding filters is done and query results need to be updated

https://www.loom.com/share/09a3c53bb17843dda3c7c75c742039aa

**Pull request merged**